src/CoreWCF.RabbitMQ/tests/IntegrationTests.cs

```diff
@@ -18,46 +21,198 @@ namespace CoreWCF.RabbitMQ.Tests
+
+                var resolver = new DependencyResolverHelper(host);
+                var testService = resolver.GetService<TestService>();
+                Assert.True(testService.ManualResetEvent.Wait(System.TimeSpan.FromSeconds(5)));
+            }
+        }
+
+        // Automated tests use a Linux container to host RabbitMQ so this test is Linux-only
+        [Fact]
+        [Trait("Category", "LinuxOnly")]
+        public void DefaultClassicQueueConfiguration_ReceiveMessage_Success()
+        {
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<DefaultClassicQueueStartup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                MessageQueueHelper.SendMessageToQueue(DefaultClassicQueueStartup.ConnectionSettings);
+
+                var resolver = new DependencyResolverHelper(host);
+                var testService = resolver.GetService<TestService>();
+                Assert.True(testService.ManualResetEvent.Wait(System.TimeSpan.FromSeconds(5)));
+            }
+        }
+
+        // Automated tests use a Linux container to host RabbitMQ so this test is Linux-only
+        [Fact]
+        [Trait("Category", "LinuxOnly")]
+        public void DefaultQuorumQueueConfiguration_ReceiveMessage_Success()
+        {
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<DefaultQuorumQueueStartup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                MessageQueueHelper.SendMessageToQueue(DefaultQuorumQueueStartup.ConnectionSettings);
+
+                var resolver = new DependencyResolverHelper(host);
+                var testService = resolver.GetService<TestService>();
+                Assert.True(testService.ManualResetEvent.Wait(System.TimeSpan.FromSeconds(5)));
+            }
+        }
+
+        // Automated tests use a Linux container to host RabbitMQ so this test is Linux-only
+        [Fact]
+        [Trait("Category", "LinuxOnly")]
+        public void DefaultQueueConfiguration_ReceiveMessage_Success()
+        {
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<DefaultQueueStartup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                MessageQueueHelper.SendMessageToQueue(DefaultQueueStartup.ConnectionSettings);
+
                 var resolver = new DependencyResolverHelper(host);
                 var testService = resolver.GetService<TestService>();
                 Assert.True(testService.ManualResetEvent.Wait(System.TimeSpan.FromSeconds(5)));
             }
         }
     }
 
-    public class Startup
+    public class ClassicQueueWithTLSStartup
+    {
+        public static Uri Uri = new("net.amqps://HOST:PORT/amq.direct/QUEUE_NAME#ROUTING_KEY");
+        private static readonly ICredentials s_credentials = new NetworkCredential(ConnectionFactory.DefaultUser, ConnectionFactory.DefaultPass);
```

Comment:
Did you mean to leave HOST and PORT not populated with real values?

Reply:
Yes, this startup class is used by a test that targets a TLS-enabled RabbitMQ host, and this test is skipped because our test service container is not configured to use that connection type.